File: lyedu-api/src/main/java/com/lyedu/entity/CertificateTemplate.java
package com.lyedu.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 证书模板实体
 *
 * @author LyEdu Team
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class CertificateTemplate extends BaseEntity {

    /** 模板名称 */
    private String name;
    /** 说明 */
    private String description;
    /** 模板配置 JSON（占位符、样式等） */
    private String config;
    /** 排序 */
    private Integer sort;
    /** 状态：0-禁用，1-启用 */
    private Integer status;
}

File: lyedu-admin/src/api/certificateTemplate.ts
import request from '@/utils/request'

export interface CertificateTemplate {
  id: number
  name: string
  description?: string
  config?: string
  sort?: number
  status: number
  createTime?: string
  updateTime?: string
}

export const getTemplateList = () => {
  return request.get<CertificateTemplate[]>('/certificate-template/list')
}

export const getTemplateById = (id: number) => {
  return request.get<CertificateTemplate>(`/certificate-template/${id}`)
}

export const createTemplate = (data: Partial<CertificateTemplate>) => {
  return request.post<number>('/certificate-template', data)
}

export const updateTemplate = (id: number, data: Partial<CertificateTemplate>) => {
  return request.put(`/certificate-template/${id}`, data)
}

export const deleteTemplate = (id: number) => {
  return request.delete(`/certificate-template/${id}`)
}
